src/solver/type_getters.jl

```diff
@@ -600,12 +600,19 @@ function get_sim_info(config::AtmosConfig)
     isnothing(restart_file) ||
         @info "Restarting simulation from file $restart_file"
     epoch = DateTime(parsed_args["start_date"], dateformat"yyyymmdd")
+    t_start_int = time_to_seconds(parsed_args["t_start"])
+    if !isnothing(restart_file) && t_start_int != 0
+        @warn "Non zero `t_start` passed with a restarting simulation. The provided `t_start` will be ignored."
+    end
     if parsed_args["use_itime"]
         dt = ITime(time_to_seconds(parsed_args["dt"]))
+        t_start = ITime(time_to_seconds(parsed_args["t_start"]), epoch = epoch)
         t_end = ITime(time_to_seconds(parsed_args["t_end"]), epoch = epoch)
-        (dt, t_end) = promote(dt, t_end)
+        # ITime(0) is added for backward compatibility (since t_start used to always be 0)
```

Comment:
How does this provide backward compatibility? If `t_start` is 0, as it was before, wouldn't this be redundant?

Reply:
This is only to ensure that types are consistent. If you see, the output of `ITime(0)` is thrown away (but it contributes to the promotion in the line before)

Reply:
Yes, because `ITime` is a parametrized type, so two `ITimes` might not be the same